src/compat/eme/get_init_data.ts

```diff
@@ -15,58 +15,96 @@
+   * event.
+   * https://www.w3.org/TR/eme-initdata-registry/
+   *
+   * `undefined` if not known.
+   */
+  type : string | undefined;
+  /** Every initialization data for that type. */
+  values: Array<{
+    /**
+     * Hex encoded system id, which identifies the key system.
+     * https://dashif.org/identifiers/content_protection/
+     *
+     * If `undefined`, we don't know the system id for that initialization data.
+     * In that case, the initialization data might even be a concatenation of
+     * the initialization data from multiple system ids.
+     */
+    systemId : string | undefined;
+    /**
+     * The initialization data itself for that type and systemId.
+     * For example, with ISOBMFF "cenc" initialization data, this will be the
+     * whole PSSH box.
+     */
+     data: Uint8Array;
+  }>;
+}
+
 /**
- * As we observed on some browsers (IE and Edge), the initialization data on
- * some segments have sometimes duplicated PSSH when sent through an encrypted
- * event (but not when pushed to the SourceBuffer).
- *
- * This function tries to guess if the initialization data contains only PSSHs
- * concatenated (as it is usually the case).
- * If that's the case, it will filter duplicated PSSHs from it.
+ * Take in input initialization data from an encrypted event and generate the
+ * corresponding array of initialization data values from it.
  *
+ * At the moment, this function only handles initialization data which have the
+ * "cenc" initialization data type.
+ * It will just return a single value with an `undefined` `systemId` for all
+ * other types of data.
  * @param {Uint8Array} initData - Raw initialization data
- * @returns {Uint8Array} - Initialization data, "cleaned"
+ * @returns {Array.<Object>}
  */
-function cleanEncryptedEvent(initData : Uint8Array) : Uint8Array {
-  let resInitData = new Uint8Array();
-  const encounteredPSSHs : Uint8Array[] = [];
-
+function getInitializationDataValues(
+  initData : Uint8Array
+) : Array<{ systemId : string | undefined; data : Uint8Array }> {
+  const result : Array<{ systemId : string | undefined; data : Uint8Array }> = [];
   let offset = 0;
+
   while (offset < initData.length) {
     if (initData.length < offset + 8 ||
         be4toi(initData, offset + 4) !== PSSH_TO_INTEGER
     ) {
       log.warn("Compat: Unrecognized initialization data. Use as is.");
-      return initData;
+      return [ { systemId: undefined,
+                 data: initData } ];
     }
 
     const len = be4toi(new Uint8Array(initData), offset);
     if (offset + len > initData.length) {
       log.warn("Compat: Unrecognized initialization data. Use as is.");
-      return initData;
+      return [ { systemId: undefined,
+                 data: initData } ];
     }
     const currentPSSH = initData.subarray(offset, offset + len);
-    if (isPSSHAlreadyEncountered(encounteredPSSHs, currentPSSH)) {
+    const systemId = getPsshSystemID(currentPSSH, 8) ?? undefined;
```

Comment:
Isn't disturbing here to set systemId to undefined if the result of the function is null?

Reply:
Why?

`getPsshSystemId` returns `null` either when:
   - the pssh version is not handled
   - the pssh seems too short to contain a systemId

And in the current function (`getInitializationDataValues`), an `undefined` systemId is documented as such
```
     * If `undefined`, we don't know the system id for that initialization data.
     * In that case, the initialization data might even be a concatenation of
     * the initialization data from multiple system ids.
```

It makes sense, no?

Reply:
After discussion, I will make `getPsshSystemId` return `undefined`

Reply:
done